src/gen.rs

```diff
@@ -165,10 +170,16 @@ fn add_to_trait_methods(
         Some(f) => format_ident!("generate_{}", f),
     };
 
+    let doc_msg = format!(
+        "Generates a custom random instance of `{}`", 
+        field_ident.as_ref().map(|x| x.to_string()).unwrap_or(ty_str.to_string())
```

Comment:
Hmm in line 168, the type name is already known, can you use that one? You can return a tuple there, because generate_ty_name has a prefix you don't want, but that saves some lines later on :)

Reply:
Oh yeah you're right, returing a tuple is much nicer